lib/codecs/tests/gelf.rs

```diff
@@ -0,0 +1,112 @@
+use bytes::Bytes;
+use chrono::{DateTime, NaiveDateTime, Utc};
+use codecs::{
+    decoding::format::gelf_fields::*, decoding::format::Deserializer, GelfDeserializerConfig,
+};
+use pretty_assertions::assert_eq;
+use serde_json::json;
+use value::Value;
+use vector_core::config::log_schema;
+
+/// Validates all the spec'd fields of GELF are deserialized.
+#[test]
+fn gelf_deserializing_all() {
+    let config = GelfDeserializerConfig;
+    let deserializer = config.build();
+
+    let add_on_int_in = "_an.add-field_int";
+    let add_on_str_in = "_an.add-field_str";
+    let add_on_int_out = "_an_add_field_int";
+    let add_on_str_out = "_an_add_field_str";
+
+    let input = json!({
+        VERSION: "1.1",
+        HOST: "example.org",
+        SHORT_MESSAGE: "A short message that helps you identify what is going on",
+        FULL_MESSAGE: "Backtrace here\n\nmore stuff",
+        TIMESTAMP: 1385053862.3072,
+        LEVEL: 1,
+        FACILITY: "foo",
+        LINE: 42,
+        FILE: "/tmp/bar",
+        add_on_int_in: 2001,
+        add_on_str_in: "A Space Odyssey",
```

Comment:
Should we allow these since the don't start with a leading `_`?

Reply:
They actually do- those fields are actually defined on lines 17-20 :)